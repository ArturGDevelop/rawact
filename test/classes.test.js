import React from "react";
import ReactDOM from "react-dom";
import * as Classes from './cases/classes';
import renderHelper from "./renderHelper";

describe('classes', () => {
	it('renders correctly', () => {
		const { rendered } = renderHelper(<Classes.Hello name="world"/>);

		expect(rendered).toMatchSnapshot();
	});

	it('pure components protect multiple re-renders', () => {
		const { element } = renderHelper(<Classes.RenderCountingPure name="world"/>);

		ReactDOM.render(<Classes.RenderCountingPure name="world"/>, element);
		let rendered = ReactDOM.render(<Classes.RenderCountingPure name="world"/>, element);
		expect(rendered).toMatchSnapshot();

		rendered = ReactDOM.render(<Classes.RenderCountingPure name="me"/>, element);
		expect(rendered).toMatchSnapshot();
	});

	xit('unmounts', () => {
		const { element } = renderHelper(<Classes.Hello name="world"/>);

		ReactDOM.unmountComponentAtNode(element);

		expect(rendered).toMatchSnapshot();
	});

	it('unmounts when changing componnt', () => {
		const { element } = renderHelper(<Classes.Hello name="world"/>);

		ReactDOM.render(<div>No class any more</div>, element);

		expect(element).toMatchSnapshot();
	});
});
