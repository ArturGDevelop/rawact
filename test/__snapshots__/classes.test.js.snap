// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`classes pure components protect multiple re-renders 1`] = `
<p>
  Hi to 
  world
   x
  1
</p>
`;

exports[`classes pure components protect multiple re-renders 2`] = `
<p>
  Hi to 
  me
   x
  2
</p>
`;

exports[`classes renders correctly 1`] = `
<p>
  Hello 
  world
  !
</p>
`;

exports[`classes unmounts when changing componnt 1`] = `
<div>
  <div>
    No class any more
  </div>
</div>
`;
